umh-core/pkg/config/config.go

```diff
@@ -52,8 +52,8 @@ type CommunicatorConfig struct {
 
 // FSMInstanceConfig is the config for a FSM instance
 type FSMInstanceConfig struct {
-	Name            string `yaml:"name"`
-	DesiredFSMState string `yaml:"desiredState"`
+	Name            string `yaml:"name,omitempty"`
```

Comment:
maybe make FSMInstanceConfig itself omitempty, but I think the desiredState and name must be required if the other exists (so fill out all fields, or fill out none, but not fill out half of them)